OPHD/States/MapViewStateTurn.cpp

```diff
@@ -412,21 +407,22 @@ void MapViewState::updateBiowasteRecycling()
 }
 
 
-void MapViewState::countFood()
+void MapViewState::produceFood()
 {
-	mFood = 0;
+	for (auto& foodProducer : enumerateComponent<FoodProduction>())
+	{
+		foodProducer.produce();
```

Comment:
Food was earlier produced in `Agridome::think`. It is now being performed in a separate function. This could arguably be a static function in `FoodProduction`, along with global food capacity/level counting, but adding the functionality here seemed like a smaller change to the overall code structure.

Reply:
This is an acceptable change. It makes more sense for a food production facility to `produce()` than to `think()` food into existance.

Reply:
the alterantive would be to add a `think()` method to `StructureComponent` and invoke that as part of the structure update loop. I think that could be more difficult to get right with respect to ordering various phases though, so I think explicit logic like this might be better.